src/test/java/org/apache/drill_web_test_framework/web_ui/pages/OptionsPage.java

```diff
@@ -0,0 +1,244 @@
+  private WebElement stringDefaultButton;
+
+  @FindBy(css = "form[id='drill.exec.functions.cast_empty_string_to_null']>div>div>select>option[value='true']:checked")
+  private WebElement valueOfBooleanTrue;
+
+  @FindBy(css = "form[id='drill.exec.functions.cast_empty_string_to_null']>div>div>select>option[value='false']:checked")
+  private WebElement valueOfBooleanFalse;
+
+  @FindBy(css = "input[value='16777217']")
+  private WebElement updatedValueOfNumeric;
+
+  @FindBy(css = "input[value='16777216']")
+  private WebElement defaultValueOfNumeric;
+
+  @FindBy(css = "input[value='test']")
+  private WebElement updatedValueOfString;
+
+  @FindBy(css = "input[value='dir']")
+  private WebElement defaultValueOfString;
+
+  @FindBy(xpath = "//div[starts-with(text(),\"Found\")]")
+  private WebElement foundMatchesOptions;
+
+  public List<List<String>> getOptionsTable() {
+    return getTable(optionTable);
+  }
+
+  public void searchFieldSendKeys(String string) {
+    searchField.clear();
+    searchField.sendKeys(string);
+  }
+
+  public void clearSearchField() {
+    clearSearchButton.click();
+  }
+
+  public void updateBooleanOption() {
+    booleanDropdown.click();
+    booleanDropdownTrue.click();
+    updateBooleanButton.click();
+  }
+
+  public void setBooleanToDefault() {
+    waitForCondition(driver -> booleanDefaultButton.isEnabled(), 5);
+    booleanDefaultButton.click();
+  }
+
+  public void updateNumericOption(String string) {
+    numericField.clear();
+    numericField.sendKeys(string);
+    numericUpdateButton.click();
+  }
+
+  public void setNumericToDefault() {
+    numericDefaultButton.click();
+  }
+
+  public void updateStringOption(String string) {
+    stringField.clear();
+    stringField.sendKeys(string);
+    stringUpdateButton.click();
+  }
+
+  public void setStringToDefault() {
+    stringDefaultButton.click();
+  }
+
+  public void selectPlannerFilter() {
+    plannerFilter.click();
+  }
+
+  public void selectStoreFilter() {
+    storeFilter.click();
+  }
+
+  public void selectParquetFilter() {
+    parquetFilter.click();
+  }
+
+  public void selectHashAggFilter() {
+    hashaggFilter.click();
+  }
+
+  public void selectHashJoinFilter() {
+    hashjoinFilter.click();
+  }
+
+  public void sortByOption() {
+    optionSortBy.click();
+  }
+
+  public void sortByValue() {
+    valueSortBy.click();
+  }
+
+  public void sortByDescription() {
+    descriptionSortBy.click();
+  }
+
+  public void waitForValueEqualsTrue() {
+    waitForCondition(driver -> valueOfBooleanTrue.isDisplayed(), 5);
+  }
+
+  public boolean isValueEqualsTrue() {
+    return valueOfBooleanTrue.isSelected();
+  }
+
+  public boolean isValueEqualsFalse() {
+    return valueOfBooleanFalse.isEnabled();
+  }
+
+  public boolean isValueOfNumericUpdated() {
+    return updatedValueOfNumeric.isDisplayed();
+  }
+
+  public boolean isValueOfNumericDefault() {
+    return defaultValueOfNumeric.isEnabled();
```

Comment:
I think it's possible to use general methods here instead of one method per option. You can pass option name as a parameter.

Reply:
It now uses fluent wait and have more complexity